README.md

```diff
@@ -1,5 +1,591 @@
+| `username` | String | Yes      | must be registered user |
+| `password` | String | Yes      |                         |
+
+#### Example
+
+```
+  {
+    "username": "siratl",
+    "password": "1234",
+  }
+```
+
+#### Response
+
+##### 200 (ok)
+
+> no issues logging in
+
+###### Example response
+
+```
+{
+    "id":2,
+    "username":"sitatl",
+    "fullName":"Elisha Atulomah",
+    "email":null,
+    "userImgUrl":null,
+    "token":"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2Vybmk"
+}
+```
+
+##### 400 (Bad Request)
+
+```
+  {
+    errorMessage: 'missing ${itemMissing}'
+  }
+```
+
+##### 401 (Unauthorized)
+
+```
+{
+  errorMessage: "passwords don't match"
+}
+```
+
+/--------------------------------------------/ ALL USERS /-----------------------------------------/
+
+### **Get all Users**
+
+_method url_: `/api/users`
+
+_http method_: **[GET]**
+
+#### Response
+
+##### 200 (ok)
+
+###### Example response
+
+```
+[
+  {
+     "id": 2,
+     "username": "siratl",
+     "password": "$2a$12$UzYfINUnqfZh2n180pBswORvPCIrwHKp3d/MEZ69DaRxoLTYj26UG",
+     "fullName": "Elisha Atulomah",
+     "email": null,
+     "userImgUrl": null
+  },
+  {
+     "id": 3,
+     "username": "bean",
+     "password": "$2a$12$5.flIIREO8kVSwAGdL2iWO1IUKaaN7VgKN9zEX/Z7XXygBupMSQ0W",
+     "fullName": "Rowan Atkinson",
+     "email": "mrbean@gmail.com",
+     "userImgUrl": ""
+  },
+]
+```
+
+/--------------------------------------------/ SINGLE USER /---------------------------------/
+
+### **Get a single User**
+
+_method url_: `/api/users/:id`
+
+_http method_: **[GET]**
+
+#### Response
+
+##### 200 (ok)
+
+###### Example response
+
+```
+[
+  {
+    "id": 2,
+    "username": "siratl",
+    "password": "$2a$12$UzYfINUnqfZh2n180pBswORvPCIrwHKp3d/MEZ69DaRxoLTYj26UG",
+    "fullName": "Elisha Atulomah",
+    "email": null,
+    "userImgUrl": null
+  }
+]
+```
+
+/--------------------------------------------/ CREATE HABIT /-----------------------------------/
+
+### **Create a Habit**
+
+_method url_: `/api/habits`
+
+_http method_: **[POST]**
+
+#### Headers
+
+| name            | type   | required | description              |
+| --------------- | ------ | -------- | ------------------------ |
+| `Content-Type`  | String | Yes      | Must be application/json |
+| `authorization` | String | Yes      | token to Authorize user  |
+
+#### Body
+
+| name         | type    | required | description |
+| ------------ | ------- | -------- | ----------- |
+| `habitTitle` | String  | Yes      |             |
+| `completed`  | Boolean | Yes      |             |
+| `category`   | String  | Yes      |             |
+
+#### Example
+
+```
+  {
+    "habitTitle": "Run 10 miles",
+    "category": "Physical Fitness",
+  }
+```
+
+#### Response
+
+##### 201 (created)
+
+###### Example Response
+
+```
+  {
+    "id": 1,
+    "habitTitle": "Run 10 miles",
+    "category": "Physical Fitness",
+    "completed": false,
+    "userId": 2
```

Comment:
we also need a way to track the completion history for each task

Reply:
I'll add that shortly